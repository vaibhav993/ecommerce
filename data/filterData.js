const filterData = [
  {
    "by": "Color",
    "options": [ "Red", "Blue", "Green" ]
  },
  {
    "by": "Gender",
    "options": [ "Men", "Women" ]
  },
  {
    "by": "Price",
    "options": [
      { "label": "0-Rs250",  "value": "0-250" },
      { "label": "Rs251-450",  "value": "251-450" },
      { "label": "Rs450",  "value": "450" },
    ]
  },
  {
    "by": "Type",
    "options": [ "Polo", "Hoodie", "Basic" ]
  }
]

export default filterData
