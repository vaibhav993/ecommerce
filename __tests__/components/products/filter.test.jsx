import { render, screen, fireEvent } from "@testing-library/react"
import { act } from "react-dom/test-utils";

import Filter from '@/components/products/filter'
import filterData from '../../../data/filterData'

jest.mock('next/router', () => require('next-router-mock'));

describe('Filter', () => {
  const createTestProps = (props={}) => ({
    filters: filterData,
    ...props
  })

  let props = createTestProps()
  it('renders Filter', () => {
    const { container } = render(<Filter {...props}/>)
    expect(container).toMatchSnapshot()
  })

  it('- should handle search by color', async () => {
    render(<Filter {...props}/>)
    const element = screen.getByTestId('filter-group-item-checkbox-Red')
    expect(element.checked).toEqual(false)
    await act(async () => {
      fireEvent.click(element)
    })
    expect(element.checked).toEqual(true)
    await act(async () => {
      fireEvent.click(element)
    })
    expect(element.checked).toEqual(false)
  })

  it('- should handle search by Gender', async () => {
    render(<Filter {...props}/>)
    const element = screen.getByTestId('filter-group-item-checkbox-Men')
    expect(element.checked).toEqual(false)
    await act(async () => {
      fireEvent.click(element)
    })
    expect(element.checked).toEqual(true)
  })

  it('- should handle search by Price', async () => {
    render(<Filter {...props}/>)
    const element = screen.getByTestId('filter-group-item-checkbox-0-Rs250')
    expect(element.checked).toEqual(false)
    await act(async () => {
      fireEvent.click(element)
    })
    expect(element.checked).toEqual(true)
  })

  it('- should handle search by Type', async () => {
    render(<Filter {...props}/>)
    const element = screen.getByTestId('filter-group-item-checkbox-Polo')
    expect(element.checked).toEqual(false)
    await act(async () => {
      fireEvent.click(element)
    })
    expect(element.checked).toEqual(true)
  })
})