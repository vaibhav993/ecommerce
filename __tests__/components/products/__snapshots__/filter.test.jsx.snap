// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders Filter 1`] = `
<div>
  <section
    class="filter"
  >
    <div
      class="filter-item"
      data-testid="filter-group"
    >
      <span>
        Color
      </span>
      <ul>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Red"
            name="Red"
            type="checkbox"
          />
          <label>
            Red
          </label>
        </li>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Blue"
            name="Blue"
            type="checkbox"
          />
          <label>
            Blue
          </label>
        </li>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Green"
            name="Green"
            type="checkbox"
          />
          <label>
            Green
          </label>
        </li>
      </ul>
    </div>
    <div
      class="filter-item"
      data-testid="filter-group"
    >
      <span>
        Gender
      </span>
      <ul>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Men"
            name="Men"
            type="checkbox"
          />
          <label>
            Men
          </label>
        </li>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Women"
            name="Women"
            type="checkbox"
          />
          <label>
            Women
          </label>
        </li>
      </ul>
    </div>
    <div
      class="filter-item"
      data-testid="filter-group"
    >
      <span>
        Price
      </span>
      <ul>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-0-Rs250"
            name="0-Rs250"
            type="checkbox"
          />
          <label>
            0-Rs250
          </label>
        </li>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Rs251-450"
            name="Rs251-450"
            type="checkbox"
          />
          <label>
            Rs251-450
          </label>
        </li>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Rs450"
            name="Rs450"
            type="checkbox"
          />
          <label>
            Rs450
          </label>
        </li>
      </ul>
    </div>
    <div
      class="filter-item"
      data-testid="filter-group"
    >
      <span>
        Type
      </span>
      <ul>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Polo"
            name="Polo"
            type="checkbox"
          />
          <label>
            Polo
          </label>
        </li>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Hoodie"
            name="Hoodie"
            type="checkbox"
          />
          <label>
            Hoodie
          </label>
        </li>
        <li
          data-testid="filter-group-item"
        >
          <input
            data-testid="filter-group-item-checkbox-Basic"
            name="Basic"
            type="checkbox"
          />
          <label>
            Basic
          </label>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
